internal/statemachine/helper.go

```diff
@@ -579,6 +579,55 @@ func removePreseeding(rootfs string) (seededSnaps map[string]string, err error)
 	return seededSnaps, nil
 }
 
+// generateGerminateCmd creates the appropriate germinate command for the
+// values configured in the image definition yaml file
+func generateGerminateCmd(imageDefinition ImageDefinition) *exec.Cmd {
+	// determine the value for the seed-dist in the form of <archive>.<series>
+	seedDist := imageDefinition.Rootfs.Archive
+	if imageDefinition.Rootfs.Seed.SeedBranch != "" {
+		seedDist = seedDist + "." + imageDefinition.Rootfs.Seed.SeedBranch
+	}
+
+	var mirror string
+	if imageDefinition.Architecture == "amd64" {
```

Comment:
Given I've spent the last week beating livecd-rootfs up until it can build an image from a rebuild archive, perhaps I'm a little more sensitive to this need than normal :) This is probably an unusual case thought, it's probably more common that a user would want to use a local mirror just to be faster.

This line of thinking leads me to something that I find pretty awkward in livecd-rootfs builds at the moment, which is that the apt config for the build is often different from that in the image we ship (commonly, the build config points at ftpmaster.internal and only enables main and restricted and the image config points at archive.ubuntu.com and enables universe and multiverse as well). Do you have a plan for that?

Reply:
Not yet! I'll have to think on that one a bit.

Reply:
Which bit? The "use different configs for build and image"? It's extremely common (like, I think basically all builds that fall into the Ubuntu flavour only have main and restricted enabled during the build but have all components enabled in the final image?)

Reply:
wait, i'm confused on timing and requirement:

1. there is definitely a requirement to have custom apt config in resulting images. but we did not state that as part of `germinate` or `deboostrap`, more as a customization after (ex: setting cloud mirrors)
2. there is availability within launchpad-buildd now to set some apt config that is used during the `bootstrap` phase. In a general sense, my understanding is that in a launchpad-buildd context, bootstraping will use the conf of the host. in context of "should we save the host conf for bootstrapping into the resulting image" I would say "this is covered by users being able to add something to sources.list.d"

so there's something happening in launchpad-buildd that's setting up a conf that's based on the host, with some override, that is then respected by debootstrap and livecd-rootfs

Reply:
Ok, we've had a discussion away from this PR, and here are some of the thoughts:

1. at the `bootstrap` phase, it should be safe to do any $TARGET_ARCH on any $HOST_ARCH
2. removing dependencies on $HOST configs is a goal
3. ensuring that the configs being used for $ARCH are clear, preferably done in a way that is readable to users without needing to dig into code, understand the implementation specifics, etc. should be a goal
4. questions are now raised on how safe it is to do cross-arch building. This PR is _not_ going to address cross-arch building.

so out of this we should focus on having a clear config, with no host requirements in place, specifically during `germinate` and `bootstrap` phases.

Reply:
> In a general sense, my understanding is that in a launchpad-buildd context, bootstraping will use the conf of the host. 

I don't think this is true fwiw. There is code in livecd-rootfs to set the mirror during build to something different when running in-data-centre: https://git.launchpad.net/livecd-rootfs/tree/live-build/auto/config#n17 . I don't know the history of this, tbh, afaik archive.ubuntu.com is routeable just fine from inside the data centre. Maybe that wasn't always the case. 

Looking at this a bit more in livecd-rootfs I don't know how to explain the current behaviour. It may just be an accident of bugs.

But anyway I see two use cases for overriding the mirror:

1. Wanting to use a local mirror of archive.ubuntu.com for speed. Perhaps we should tell people to set up squid-deb-proxy for this, I don't know. In any case this probably shouldn't be specified in the image definition file, and we want /etc/apt/sources.list in the built image to refer to archive.ubuntu.com.
2. Wanting to use an actual different source for packages for some reason, like perhaps http://rebuild-test.ubuntu.com/ubuntu-test-rebuild-20220506-jammy-v3/ubuntu/. This isn't supported by livecd-rootfs at all today so maybe this can be ignored for now (although I might try and make my hacks into something that can be submitted soon). This _should_ be specified in the image definition file and should result in a different sources.list.

Sorry if I've derailed things a bit here.

Reply:
Oh and anyway, the other thing is which components are enabled, and when. Most images build with only main and restricted enabled, but ship with sources.list referencing all components. That's a separate thing again.